pkg/oauthserver/setup_test.go

```diff
@@ -0,0 +1,119 @@
+package oauthserver_test
+
+import (
+	"net/http"
+
+	oauthv1 "github.com/openshift/api/oauth/v1"
+	osinv1 "github.com/openshift/api/osin/v1"
+	fakeoauthclient "github.com/openshift/client-go/oauth/clientset/versioned/fake"
+	typedv1 "github.com/openshift/client-go/oauth/clientset/versioned/typed/oauth/v1"
+	bootstrap "github.com/openshift/library-go/pkg/authentication/bootstrapauthenticator"
+	"github.com/openshift/oauth-server/pkg/config"
+	"github.com/openshift/oauth-server/pkg/oauthserver"
+	"github.com/openshift/oauth-server/pkg/userregistry/identitymapper"
+	"golang.org/x/crypto/bcrypt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	fakekube "k8s.io/client-go/kubernetes/fake"
+)
+
+const (
+	// testUser == kubeadmin
+	testPassword = "testpasswordxtestpasswordx" // must have at least len 23
+
+	clientKey   = "client_id"
+	clientValue = "my_client"
+
+	responseTypeKey   = "response_type"
+	responseTypeValue = "code"
+)
+
+type BootstrapUserIDP func() (*bootstrap.BootstrapUserData, bool, error)
+
+var _ bootstrap.BootstrapUserDataGetter = (*BootstrapUserIDP)(nil)
+
+func (b BootstrapUserIDP) IsEnabled() (bool, error) {
+	return true, nil
+}
+
+func (b BootstrapUserIDP) Get() (*bootstrap.BootstrapUserData, bool, error) {
+	return b()
+}
+
+func kubeAdmin(passwordHash []byte, auth bool, err error) BootstrapUserIDP {
+	return func() (*bootstrap.BootstrapUserData, bool, error) {
+		hash, err := bcrypt.GenerateFromPassword(passwordHash, bcrypt.DefaultCost)
+		if err != nil {
+			panic(err)
+		}
+
+		return &bootstrap.BootstrapUserData{
+			PasswordHash: hash,
+			UID:          "kubeadmin-user-id",
+		}, auth, err
+	}
+}
+
+func setup() http.Handler {
+	kubeClient := fakekube.NewSimpleClientset()
+	oauthClient := goodClientRegistry(
+		clientValue,
+		[]string{"myredirect"},
+		[]string{"myscope1", "myscope2"},
+	)
+
+	kubeAdminIDP := kubeAdmin([]byte(testPassword), true, nil)
+
+	opts := osinv1.OAuthConfig{
+		LoginURL: "/oauth/login",
+		IdentityProviders: []osinv1.IdentityProvider{
+			{
+				Name:            bootstrap.BootstrapUser, // aka kube:admin
+				UseAsChallenger: false,
+				UseAsLogin:      false,
+				MappingMethod:   string(identitymapper.MappingMethodClaim), // irrelevant, but needs to be valid
+				Provider: runtime.RawExtension{
+					Object: &config.BootstrapIdentityProvider{},
+				},
+			},
+		},
+		GrantConfig: osinv1.GrantConfig{
+			Method: osinv1.GrantHandlerAuto,
+		},
+	}
+
+	oasc := oauthserver.OAuthServerConfig{
```

Comment:
`s/oasc/serverConfig`

Reply:
kk

Reply:
hm, if it has behavior, it is not a config anymore. What about server? :smile:

Reply:
changed it to a different name, maybe you won't check my comments here and you are just fine that I changed it...